In [1]:
# from google.colab import drive
# drive.mount('/content/gdrive')
path_initial = "C:/Users/Richa/Desktop/upgrad/InvestmentCaseStudy/"

### Importing required Libraries

In [2]:
import pandas as pd
import numpy as np
import sys

### Checkpoint-1 Reading `companies` and `rounds2` dataset<br>
-    We are overriding default encoding **`utf-8`**  with **`latin`** to read the dataframe properly

In [3]:
companies = pd.read_csv(path_initial + 'companies.txt', sep = '\t', encoding='latin')
rounds2 = pd.read_csv(path_initial + 'rounds2.csv', encoding='latin')

### Let us see the data to get an idea of available columns and how the data looks

In [4]:
companies.head()

permalink                    name  \
0                  /Organization/-Fame                   #fame   
1               /Organization/-Qounter                :Qounter   
2  /Organization/-The-One-Of-Them-Inc-  (THE) ONE of THEM,Inc.   
3                /Organization/0-6-Com                 0-6.com   
4       /Organization/004-Technologies        004 Technologies   

                        homepage_url  \
0                 http://livfame.com   
1             http://www.qounter.com   
2                http://oneofthem.jp   
3                 http://www.0-6.com   
4  http://004gmbh.de/en/004-interact   

                                       category_list     status country_code  \
0                                              Media  operating          IND   
1  Application Platforms|Real Time|Social Network...  operating          USA   
2                                  Apps|Games|Mobile  operating          NaN   
3                                        Curated Web  operating          CHN   
4                                           Software  operating          USA   

  state_code                 region           city  founded_at  
0         16                 Mumbai         Mumbai         NaN  
1         DE             DE - Other  Delaware City  04-09-2014  
2        NaN                    NaN            NaN         NaN  
3         22                Beijing        Beijing  01-01-2007  
4         IL  Springfield, Illinois      Champaign  01-01-2010

In [5]:
companies.tail()

permalink  \
66363  /Organization/Zznode-Science-And-Technology-Co...   
66364                          /Organization/Zzzzapp-Com   
66365                               /Organization/ÁEron   
66366                             /Organization/ÔAsys-2   
66367  /Organization/Ä°Novatiff-Reklam-Ve-Tanä±Tä±M-H...   

                                                name            homepage_url  \
66363                  ZZNode Science and Technology   http://www.zznode.com   
66364                          Zzzzapp Wireless ltd.  http://www.zzzzapp.com   
66365                                         ÁERON    http://www.aeron.hu/   
66366                                         Ôasys    http://www.oasys.io/   
66367  İnovatiff Reklam ve Tanıtım Hizmetleri Tic    http://inovatiff.com   

                                           category_list     status  \
66363                                Enterprise Software  operating   
66364        Advertising|Mobile|Web Development|Wireless  operating   
66365                                                NaN  operating   
66366  Consumer Electronics|Internet of Things|Teleco...  operating   
66367                 Consumer Goods|E-Commerce|Internet  operating   

      country_code state_code       region           city  founded_at  
66363          CHN         22      Beijing        Beijing         NaN  
66364          HRV         15        Split          Split  13-05-2012  
66365          NaN        NaN          NaN            NaN  01-01-2011  
66366          USA         CA  SF Bay Area  San Francisco  01-01-2014  
66367          NaN        NaN          NaN            NaN         NaN

In [6]:
rounds2.head()

company_permalink  \
0                  /organization/-fame   
1               /ORGANIZATION/-QOUNTER   
2               /organization/-qounter   
3  /ORGANIZATION/-THE-ONE-OF-THEM-INC-   
4                /organization/0-6-com   

                           funding_round_permalink funding_round_type  \
0  /funding-round/9a01d05418af9f794eebff7ace91f638            venture   
1  /funding-round/22dacff496eb7acb2b901dec1dfe5633            venture   
2  /funding-round/b44fbb94153f6cdef13083530bb48030               seed   
3  /funding-round/650b8f704416801069bb178a1418776b            venture   
4  /funding-round/5727accaeaa57461bd22a9bdd945382d            venture   

  funding_round_code   funded_at  raised_amount_usd  
0                  B  05-01-2015         10000000.0  
1                  A  14-10-2014                NaN  
2                NaN  01-03-2014           700000.0  
3                  B  30-01-2014          3406878.0  
4                  A  19-03-2008          2000000.0

In [7]:
rounds2.tail()

company_permalink  \
114944                          /organization/zzzzapp-com   
114945                          /ORGANIZATION/ZZZZAPP-COM   
114946                               /organization/ãeron   
114947                             /ORGANIZATION/ÔASYS-2   
114948  /organization/ä°novatiff-reklam-ve-tanä±tä±m-h...   

                                funding_round_permalink funding_round_type  \
114944  /funding-round/8f6d25b8ee4199e586484d817bceda05   convertible_note   
114945  /funding-round/ff1aa06ed5da186c84f101549035d4ae               seed   
114946  /funding-round/59f4dce44723b794f21ded3daed6e4fe            venture   
114947  /funding-round/35f09d0794651719b02bbfd859ba9ff5               seed   
114948  /funding-round/af942869878d2cd788ef5189b435ebc4              grant   

       funding_round_code   funded_at  raised_amount_usd  
114944                NaN  01-03-2014            41313.0  
114945                NaN  01-05-2013            32842.0  
114946                  A  01-08-2014                NaN  
114947                NaN  01-01-2015            18192.0  
114948                NaN  01-10-2013            14851.0

### So we can notice special characters in `permalink` and `name` columns in `companies` dataframe in tail section
### And same thing for `company_permalink` columns in `rounds2` dataframe
- Below logic is added to handle special characters

In [8]:
companies_columns_to_remove_special_char = ['permalink','name']
rounds2_columns_to_remove_special_char = ['company_permalink']

companies[companies_columns_to_remove_special_char] = companies[companies_columns_to_remove_special_char].applymap(lambda x 
                                                                                               : str(x).encode('utf-8').decode('ascii','ignore'))
rounds2[rounds2_columns_to_remove_special_char] = rounds2[rounds2_columns_to_remove_special_char].applymap(lambda x 
                                                                                               : str(x).encode('utf-8').decode('ascii','ignore'))

### Now let's see tail section of both the dataframes to check presence of special characters

In [9]:
companies.tail()

permalink  \
66363  /Organization/Zznode-Science-And-Technology-Co...   
66364                          /Organization/Zzzzapp-Com   
66365                                 /Organization/Eron   
66366                               /Organization/Asys-2   
66367  /Organization/Novatiff-Reklam-Ve-TanTM-Hizmetl...   

                                          name            homepage_url  \
66363            ZZNode Science and Technology   http://www.zznode.com   
66364                    Zzzzapp Wireless ltd.  http://www.zzzzapp.com   
66365                                     ERON    http://www.aeron.hu/   
66366                                     asys    http://www.oasys.io/   
66367  novatiff Reklam ve Tantm Hizmetleri Tic    http://inovatiff.com   

                                           category_list     status  \
66363                                Enterprise Software  operating   
66364        Advertising|Mobile|Web Development|Wireless  operating   
66365                                                NaN  operating   
66366  Consumer Electronics|Internet of Things|Teleco...  operating   
66367                 Consumer Goods|E-Commerce|Internet  operating   

      country_code state_code       region           city  founded_at  
66363          CHN         22      Beijing        Beijing         NaN  
66364          HRV         15        Split          Split  13-05-2012  
66365          NaN        NaN          NaN            NaN  01-01-2011  
66366          USA         CA  SF Bay Area  San Francisco  01-01-2014  
66367          NaN        NaN          NaN            NaN         NaN

In [10]:
rounds2.tail()

company_permalink  \
114944                          /organization/zzzzapp-com   
114945                          /ORGANIZATION/ZZZZAPP-COM   
114946                                 /organization/eron   
114947                               /ORGANIZATION/ASYS-2   
114948  /organization/novatiff-reklam-ve-tantm-hizmetl...   

                                funding_round_permalink funding_round_type  \
114944  /funding-round/8f6d25b8ee4199e586484d817bceda05   convertible_note   
114945  /funding-round/ff1aa06ed5da186c84f101549035d4ae               seed   
114946  /funding-round/59f4dce44723b794f21ded3daed6e4fe            venture   
114947  /funding-round/35f09d0794651719b02bbfd859ba9ff5               seed   
114948  /funding-round/af942869878d2cd788ef5189b435ebc4              grant   

       funding_round_code   funded_at  raised_amount_usd  
114944                NaN  01-03-2014            41313.0  
114945                NaN  01-05-2013            32842.0  
114946                  A  01-08-2014                NaN  
114947                NaN  01-01-2015            18192.0  
114948                NaN  01-10-2013            14851.0

### Now we can see those special characters are removed

### Now let us get an idea about the shape and structure of `companies` dataframe

In [11]:
companies.shape

(66368, 10)

In [12]:
companies.describe()

permalink   name                homepage_url category_list  \
count                66368  66368                       61310         63220   
unique               66368  66103                       61191         27296   
top     /Organization/Teak  Karma  http://www.askforoffer.com      Software   
freq                     1      4                           5          3995   

           status country_code state_code       region           city  \
count       66368        59410      57821        58338          58340   
unique          4          137        311         1092           5111   
top     operating          USA         CA  SF Bay Area  San Francisco   
freq        53034        37601      12900         8804           3526   

        founded_at  
count        51147  
unique        3978  
top     01-01-2012  
freq          2730

In [13]:
companies.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 66368 entries, 0 to 66367
Data columns (total 10 columns):
permalink        66368 non-null object
name             66368 non-null object
homepage_url     61310 non-null object
category_list    63220 non-null object
status           66368 non-null object
country_code     59410 non-null object
state_code       57821 non-null object
region           58338 non-null object
city             58340 non-null object
founded_at       51147 non-null object
dtypes: object(10)
memory usage: 5.1+ MB


In [14]:
rounds2.shape

(114949, 6)

In [15]:
rounds2.describe()

raised_amount_usd
count       9.495900e+04
mean        1.042687e+07
std         1.148212e+08
min         0.000000e+00
25%         3.225000e+05
50%         1.680511e+06
75%         7.000000e+06
max         2.127194e+10

In [16]:
rounds2.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 114949 entries, 0 to 114948
Data columns (total 6 columns):
company_permalink          114949 non-null object
funding_round_permalink    114949 non-null object
funding_round_type         114949 non-null object
funding_round_code         31140 non-null object
funded_at                  114949 non-null object
raised_amount_usd          94959 non-null float64
dtypes: float64(1), object(5)
memory usage: 5.3+ MB


 <font color='green'>From `companies.describe()` command we can get an idea that `permalink` might be the right canddate to uniquely identify companies in `companies` df</font>
- Because out of 66368 records all are unique
- But to be sure we need to check in case insentivie way
- Also by seeing column names it looks like apart from `permalink` , `name` or `homepage_url` might be used to identify companies uniquely
- Let us confirm this


In [17]:
# companies df duplicate count case-insensitive

companies_columnss_to_check_duplicates = ['permalink','name','homepage_url']
pd.DataFrame(list(map(lambda x : 
                      (x, np.sum(companies[x].str.upper().duplicated())), 
                      companies_columnss_to_check_duplicates)), 
             columns=['col_name', 'duplicate_count'])

col_name  duplicate_count
0     permalink                0
1          name              330
2  homepage_url             5180

In [18]:
# rounds2 df duplicate count case-insensitive

rounds2_columnss_to_check_duplicates = ['company_permalink','funding_round_permalink']
pd.DataFrame(list(map(lambda x : 
                      (x, np.sum(rounds2[x].str.upper().duplicated())), 
                      rounds2_columnss_to_check_duplicates)), 
             columns=['col_name', 'duplicate_count'])

col_name  duplicate_count
0        company_permalink            48581
1  funding_round_permalink                0

- ###  <font color='green'> From above 2 commands it is clearly evident that </font>
  
 1.    Unique Identifier for `companies` is `permalink`

 2.    Unique identifier for `rounds2` is `funding_round_permalink`

### Unique companies present in `rounds2`

In [19]:
np.sum(rounds2['company_permalink'].str.upper().duplicated()==False)

66368

### Unique companies present in `companies`

In [20]:
np.sum(companies['permalink'].str.upper().duplicated()==False)

66368

### Companies present in `rounds2` df but not in `companies` df

In [21]:
set(rounds2.company_permalink.str.upper()) - set(companies.permalink.str.upper())

set()

### Let us see Companies present in `companies df` df but not in `rounds2`

In [22]:
set(companies.permalink.str.upper()) - set(rounds2.company_permalink.str.upper()) 

set()

- ### <font color='green'>From the above result we can see both the dataframes have same companies</font>

### Let us check if removing special characters from dataframes made any impact or not

In [23]:
companies_test = pd.read_csv(path_initial + 'companies.txt', sep = '\t', encoding='latin')
rounds2_test = pd.read_csv(path_initial + 'rounds2.csv', encoding='latin')

set(companies_test.permalink.str.upper()) - set(rounds2_test.company_permalink.str.upper()) 

{'/ORGANIZATION/E-CÃ\x9aBICA',
 '/ORGANIZATION/ENERGYSTONE-GAMES-Ç\x81ΜÇŸ³Æ¸¸Æ\x88\x8f',
 '/ORGANIZATION/HUIZUCHE-COM-Æ\x83\xa0ǧ\x9fȽ¦',
 '/ORGANIZATION/TIPCAT-INTERACTIVE-Ʋ\x99È\x88\x9fÄ¿¡Æ\x81¯Ç§\x91Æ\x9a\x80',
 '/ORGANIZATION/ZENGAME-Ǧ\x85Ƹ¸Ç§\x91Æ\x9a\x80'}

- ### So we can conclude handling the special characters impacted our analysis in a positive way

### Before merging we need to make the joining column in uppercase as pandas unlike sql does not join in a case insenitive way

In [24]:
companies.permalink = companies.permalink.str.upper()

In [25]:
rounds2.company_permalink = rounds2.company_permalink.str.upper()

In [26]:
companies.head()

permalink                    name  \
0                  /ORGANIZATION/-FAME                   #fame   
1               /ORGANIZATION/-QOUNTER                :Qounter   
2  /ORGANIZATION/-THE-ONE-OF-THEM-INC-  (THE) ONE of THEM,Inc.   
3                /ORGANIZATION/0-6-COM                 0-6.com   
4       /ORGANIZATION/004-TECHNOLOGIES        004 Technologies   

                        homepage_url  \
0                 http://livfame.com   
1             http://www.qounter.com   
2                http://oneofthem.jp   
3                 http://www.0-6.com   
4  http://004gmbh.de/en/004-interact   

                                       category_list     status country_code  \
0                                              Media  operating          IND   
1  Application Platforms|Real Time|Social Network...  operating          USA   
2                                  Apps|Games|Mobile  operating          NaN   
3                                        Curated Web  operating          CHN   
4                                           Software  operating          USA   

  state_code                 region           city  founded_at  
0         16                 Mumbai         Mumbai         NaN  
1         DE             DE - Other  Delaware City  04-09-2014  
2        NaN                    NaN            NaN         NaN  
3         22                Beijing        Beijing  01-01-2007  
4         IL  Springfield, Illinois      Champaign  01-01-2010

In [27]:
rounds2.head()

company_permalink  \
0                  /ORGANIZATION/-FAME   
1               /ORGANIZATION/-QOUNTER   
2               /ORGANIZATION/-QOUNTER   
3  /ORGANIZATION/-THE-ONE-OF-THEM-INC-   
4                /ORGANIZATION/0-6-COM   

                           funding_round_permalink funding_round_type  \
0  /funding-round/9a01d05418af9f794eebff7ace91f638            venture   
1  /funding-round/22dacff496eb7acb2b901dec1dfe5633            venture   
2  /funding-round/b44fbb94153f6cdef13083530bb48030               seed   
3  /funding-round/650b8f704416801069bb178a1418776b            venture   
4  /funding-round/5727accaeaa57461bd22a9bdd945382d            venture   

  funding_round_code   funded_at  raised_amount_usd  
0                  B  05-01-2015         10000000.0  
1                  A  14-10-2014                NaN  
2                NaN  01-03-2014           700000.0  
3                  B  30-01-2014          3406878.0  
4                  A  19-03-2008          2000000.0

- ### Now merge the 2 dataframes to create `master_frame`
- We can do inner join safely as both dataframes has same number of unique companies and their difference set has 0 element as seen above

In [28]:
master_frame = companies.merge(rounds2, how='inner', left_on=['permalink'], right_on=['company_permalink'])
master_frame.head()

permalink                    name  \
0                  /ORGANIZATION/-FAME                   #fame   
1               /ORGANIZATION/-QOUNTER                :Qounter   
2               /ORGANIZATION/-QOUNTER                :Qounter   
3  /ORGANIZATION/-THE-ONE-OF-THEM-INC-  (THE) ONE of THEM,Inc.   
4                /ORGANIZATION/0-6-COM                 0-6.com   

             homepage_url                                      category_list  \
0      http://livfame.com                                              Media   
1  http://www.qounter.com  Application Platforms|Real Time|Social Network...   
2  http://www.qounter.com  Application Platforms|Real Time|Social Network...   
3     http://oneofthem.jp                                  Apps|Games|Mobile   
4      http://www.0-6.com                                        Curated Web   

      status country_code state_code      region           city  founded_at  \
0  operating          IND         16      Mumbai         Mumbai         NaN   
1  operating          USA         DE  DE - Other  Delaware City  04-09-2014   
2  operating          USA         DE  DE - Other  Delaware City  04-09-2014   
3  operating          NaN        NaN         NaN            NaN         NaN   
4  operating          CHN         22     Beijing        Beijing  01-01-2007   

                     company_permalink  \
0                  /ORGANIZATION/-FAME   
1               /ORGANIZATION/-QOUNTER   
2               /ORGANIZATION/-QOUNTER   
3  /ORGANIZATION/-THE-ONE-OF-THEM-INC-   
4                /ORGANIZATION/0-6-COM   

                           funding_round_permalink funding_round_type  \
0  /funding-round/9a01d05418af9f794eebff7ace91f638            venture   
1  /funding-round/22dacff496eb7acb2b901dec1dfe5633            venture   
2  /funding-round/b44fbb94153f6cdef13083530bb48030               seed   
3  /funding-round/650b8f704416801069bb178a1418776b            venture   
4  /funding-round/5727accaeaa57461bd22a9bdd945382d            venture   

  funding_round_code   funded_at  raised_amount_usd  
0                  B  05-01-2015         10000000.0  
1                  A  14-10-2014                NaN  
2                NaN  01-03-2014           700000.0  
3                  B  30-01-2014          3406878.0  
4                  A  19-03-2008          2000000.0

#### Below command will correct the display of  the numeric field 

In [29]:
pd.options.display.float_format = "{:.2f}".format

In [30]:
master_frame.shape

(114949, 16)

In [31]:
master_frame.describe()

raised_amount_usd
count           94959.00
mean         10426869.33
std         114821247.98
min                 0.00
25%            322500.00
50%           1680511.00
75%           7000000.00
max       21271935000.00

In [32]:
master_frame.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 114949 entries, 0 to 114948
Data columns (total 16 columns):
permalink                  114949 non-null object
name                       114949 non-null object
homepage_url               108815 non-null object
category_list              111539 non-null object
status                     114949 non-null object
country_code               106271 non-null object
state_code                 104003 non-null object
region                     104782 non-null object
city                       104785 non-null object
founded_at                 94428 non-null object
company_permalink          114949 non-null object
funding_round_permalink    114949 non-null object
funding_round_type         114949 non-null object
funding_round_code         31140 non-null object
funded_at                  114949 non-null object
raised_amount_usd          94959 non-null float64
dtypes: float64(1), object(15)
memory usage: 14.9+ MB


- ### Now  let us verify the missing / Null /NaN values and check if we can do anything

In [33]:
round(100*master_frame.isna().mean(),2)

permalink                  0.00
name                       0.00
homepage_url               5.34
category_list              2.97
status                     0.00
country_code               7.55
state_code                 9.52
region                     8.84
city                       8.84
founded_at                17.85
company_permalink          0.00
funding_round_permalink    0.00
funding_round_type         0.00
funding_round_code        72.91
funded_at                  0.00
raised_amount_usd         17.39
dtype: float64

- ### Checkpoint 2-Funding Type Analysis

- ### As our analysis is based on Investment generated, we will focus on raised_amount_usd

- ### Out of ths 17.39% missing values in `raised_amount_usd` let us see share by `funding_round_type`



In [34]:
total_funding_type_count = pd.DataFrame(master_frame.groupby('funding_round_type',sort=True)['funding_round_type'].count())

missing_fund_value_for_type_count = master_frame[master_frame.raised_amount_usd.isna()].groupby('funding_round_type',sort=True)['funding_round_type'].count()

percentage_missing = round(100*(master_frame[master_frame.raised_amount_usd.isna()].groupby('funding_round_type',sort=True)['funding_round_type'].count())/(master_frame.groupby('funding_round_type',sort=True)['funding_round_type'].count()),2)

z = pd.concat([total_funding_type_count,missing_fund_value_for_type_count,percentage_missing],axis=1)

z.columns=['Total_Records','Missing_Fund_Value','Missing_Percentage']
z

C:\Users\Richa\OneDrive\Documents\New folder\lib\site-packages\ipykernel_launcher.py:7: FutureWarning: Sorting because non-concatenation axis is not aligned. A future version
of pandas will change to not sort by default.

To accept the future behavior, pass 'sort=False'.

To retain the current behavior and silence the warning, pass 'sort=True'.

  import sys


Total_Records  Missing_Fund_Value  Missing_Percentage
angel                           6094             1234.00               20.25
convertible_note                1817              370.00               20.36
debt_financing                  6895              213.00                3.09
equity_crowdfunding             3257             2029.00               62.30
grant                           2200               76.00                3.45
non_equity_assistance            191              118.00               61.78
post_ipo_debt                    152                 nan                 nan
post_ipo_equity                  638               10.00                1.57
private_equity                  2285              349.00               15.27
product_crowdfunding             410               24.00                5.85
secondary_market                  95               65.00               68.42
seed                           30524             6921.00               22.67
undisclosed                     4897             3315.00               67.69
venture                        55494             5266.00                9.49


1.   venture - 9.49%
2.   angel - 20.25%
3.   seed - 22.67%
4.   private_equity - 15.27%

So we can see some dataquality issue here, There are more than 20% missing value in some of the funding type.

Though this indicates data quality issue, to give better suggestion to Spark Funds we should impute them



In [35]:
pd.options.display.float_format = "{:.2f}".format

- ### Describe() command ignores missing and NAN value while calculating mean , count ,Std, min for numeric fields so we don't need to impute  those rows where raised_amount_usd is 0 for funding type analyis

In [36]:
##Calcualte average for funding type 

master_frame['raised_amount_usd'].groupby(master_frame['funding_round_type']).describe()


count         mean          std      min        25%  \
funding_round_type                                                             
angel                  4860.00    958694.47   7404397.12     0.00  150000.00   
convertible_note       1447.00   1453438.54   8043846.14     0.00   78847.50   
debt_financing         6682.00  17043526.02 280320356.61     0.00  330484.25   
equity_crowdfunding    1228.00    538368.21   1719329.42     0.00    7000.00   
grant                  2124.00   4300576.34  25587424.15     0.00   50000.00   
non_equity_assistance    73.00    411203.05   1771747.25     0.00   25000.00   
post_ipo_debt           152.00 168704571.82 602473804.43 70000.00 4000000.00   
post_ipo_equity         628.00  82182493.87 469188915.85    77.00 3764794.00   
private_equity         1936.00  73308593.03 198113458.41     0.00 5000000.00   
product_crowdfunding    386.00   1363131.07   4698379.78     0.00   40241.75   
secondary_market         30.00  79649630.10 140103760.22  2323.00 5025000.00   
seed                  23603.00    719818.00   2221732.80     0.00   55000.00   
undisclosed            1582.00  19242370.23 199323809.16     0.00  220000.00   
venture               50228.00  11748949.13  86352066.56     0.00 1600901.75   

                              50%         75%            max  
funding_round_type                                            
angel                   400000.00  1000000.00   494511992.00  
convertible_note        272000.00  1000000.00   185048285.00  
debt_financing         1100000.00  4000000.00 21271935000.00  
equity_crowdfunding     100000.00   395901.00    27400000.00  
grant                   201684.00  1700000.00   750000000.00  
non_equity_assistance    60000.00   125000.00    15000000.00  
post_ipo_debt         19950000.00 60000000.00  5800000000.00  
post_ipo_equity       12262852.50 40000000.00 10664943639.00  
private_equity        20000000.00 75000000.00  4745460219.00  
product_crowdfunding    183915.00   735000.00    52000000.00  
secondary_market      32600000.00 84475000.00   680611554.00  
seed                    275000.00  1000000.00   200000000.00  
undisclosed            1018680.00  4848804.00  5000000000.00  
venture                5000000.00 12000000.00 17600000000.00


- ### Considering that Spark Funds wants to invest between 5 to 15 million USD per investment round

- ### Calculating investment type which is the most suitable


In [37]:
average_ft= master_frame['raised_amount_usd'].groupby(master_frame['funding_round_type']).mean()
average_ft.sort_values(ascending = False) 
most_suitable_ft = average_ft[average_ft.between(5000000,15000000) ].index[0]
most_suitable_ft

'venture'

In [38]:
master_frame.loc[master_frame.category_list.isna()].head()

permalink         name            homepage_url  \
99   /ORGANIZATION/123FENG-COM  123Feng.Com     http://123feng.com/   
123        /ORGANIZATION/15FEN        15Fen   http://www.15fen.com/   
133     /ORGANIZATION/17-MEDIA     17 Media         http://17.media   
181      /ORGANIZATION/1MXIANS      1mxians  http://www.1mxian.com/   
185    /ORGANIZATION/1RP-MEDIA    1RP Media                     NaN   

    category_list     status country_code state_code       region  \
99            NaN  operating          CHN          2     Hangzhou   
123           NaN  operating          CHN         30    Guangzhou   
133           NaN  operating          NaN        NaN          NaN   
181           NaN  operating          CHN         22      Beijing   
185           NaN  operating          USA         CA  Los Angeles   

            city  founded_at          company_permalink  \
99      Hangzhou  01-01-2014  /ORGANIZATION/123FENG-COM   
123    Guangzhou         NaN        /ORGANIZATION/15FEN   
133          NaN  15-07-2015     /ORGANIZATION/17-MEDIA   
181      Beijing  01-01-2014      /ORGANIZATION/1MXIANS   
185  Los Angeles  01-01-2011    /ORGANIZATION/1RP-MEDIA   

                             funding_round_permalink funding_round_type  \
99   /funding-round/9d83084d87dc63a309c9a6fe9bf59d1b            venture   
123  /funding-round/b660e2c345e5cda7db2f1e0176a450b3               seed   
133  /funding-round/f8ffde9fa822843ac6ea9b66cc615df1            venture   
181  /funding-round/6950448f81c6706abe61f659e15a7c87            venture   
185  /funding-round/f7400a94939ef73f70af0c52b54b31ce               seed   

    funding_round_code   funded_at  raised_amount_usd  
99                   A  27-10-2015        13000000.00  
123                NaN  15-12-2013                nan  
133                  A  05-11-2015        10000000.00  
181                  A  02-12-2015        15000000.00  
185                NaN  29-06-2012                nan

In [39]:
#Get all the unique funding type 
pd.unique(master_frame.funding_round_type)

array(['venture', 'seed', 'undisclosed', 'equity_crowdfunding',
       'convertible_note', 'private_equity', 'debt_financing', 'angel',
       'grant', 'secondary_market', 'post_ipo_equity', 'post_ipo_debt',
       'product_crowdfunding', 'non_equity_assistance'], dtype=object)

- ### Checkpoint 3-Country Analysis 

In [40]:
# Read english speaking countries csv data

countries = pd.read_csv(path_initial + 'Country_Codes.csv')
# countries.head()
countries

Country Code  ENG
0                           Botswana  BWA  Yes
1                           Cameroon  CMR  Yes
2                           Ethiopia  ETH  Yes
3                            Eritrea  ERI  Yes
4                         The Gambia  GMB  Yes
5                              Ghana  GHA  Yes
6                              Kenya  KEN  Yes
7                            Lesotho  LSO  Yes
8                            Liberia  LBR  Yes
9                             Malawi  MWI  Yes
10                         Mauritius  MUS  Yes
11                           Namibia  NAM  Yes
12                           Nigeria  NGA  Yes
13                            Rwanda  RWA  Yes
14                        Seychelles  SYC  Yes
15                      Sierra Leone  SLE  Yes
16                      South Africa  ZAF  Yes
17                       South Sudan  SSD  Yes
18                             Sudan  SDN  Yes
19                         Swaziland  SWZ  Yes
20                          Tanzania  TZA  Yes
21                            Uganda  UGA  Yes
22                            Zambia  ZMB  Yes
23                          Zimbabwe  ZWE  Yes
24               Antigua and Barbuda  ATG  Yes
25                       The Bahamas  BHS  Yes
26                          Barbados  BRB  Yes
27                            Belize  BLZ  Yes
28                            Canada  CAN  Yes
29                          Dominica  DMA  Yes
30                           Grenada  GRD  Yes
31                            Guyana  GUY  Yes
32                           Jamaica  JAM  Yes
33             Saint Kitts and Nevis  KNA  Yes
34                       Saint Lucia  LCA  Yes
35  Saint Vincent and the Grenadines  VCT  Yes
36               Trinidad and Tobago  TTO  Yes
37                     United States  USA  Yes
38                             India  IND  Yes
39                          Pakistan  PAK  Yes
40                       Philippines  PHL  Yes
41                         Singapore  SGP  Yes
42                         Australia  AUS  Yes
43                              Fiji  FJI  Yes
44                          Kiribati  KIR  Yes
45                  Marshall Islands  MHL  Yes
46    Federated States of Micronesia  FSM  Yes
47                             Nauru  NRU  Yes
48                       New Zealand  NZL  Yes
49                             Palau  PLW  Yes
50                  Papua New Guinea  PNG  Yes
51                             Samoa  WSM  Yes
52                   Solomon Islands  SLB  Yes
53                             Tonga  TON  Yes
54                            Tuvalu  TUV  Yes
55                           Vanuatu  VUT  Yes
56                           Ireland  IRL  Yes
57                             Malta  MLT  Yes
58                    United Kingdom  GBR  Yes

In [41]:
## lets see in how many record counrty is blank
round(100*master_frame.isna().mean(),2)

permalink                  0.00
name                       0.00
homepage_url               5.34
category_list              2.97
status                     0.00
country_code               7.55
state_code                 9.52
region                     8.84
city                       8.84
founded_at                17.85
company_permalink          0.00
funding_round_permalink    0.00
funding_round_type         0.00
funding_round_code        72.91
funded_at                  0.00
raised_amount_usd         17.39
dtype: float64

- ### As there are some rows where county is blank so we will apply inner join to remove those rows 

In [42]:
master_frame = master_frame.merge(countries[['Country','Code']], how='inner', left_on='country_code', right_on='Code')



In [43]:
## Check master_frame 
master_frame.head(5)

permalink           name  \
0                 /ORGANIZATION/-FAME          #fame   
1             /ORGANIZATION/1000LOOKZ      1000Lookz   
2                /ORGANIZATION/1CLICK         1CLICK   
3      /ORGANIZATION/21DIAMONDS-INDIA     21Diamonds   
4  /ORGANIZATION/247-LEARNING-PRIVATE  24x7 Learning   

                  homepage_url             category_list     status  \
0           http://livfame.com                     Media  operating   
1         http://1000lookz.com                    Beauty  operating   
2             http://1click.io               Chat|Mobile   acquired   
3     http://www.21diamonds.de                E-Commerce  operating   
4  http://www.24x7learning.com  EdTech|Education|Systems  operating   

  country_code state_code     region       city  founded_at  \
0          IND         16     Mumbai     Mumbai         NaN   
1          IND         25    Chennai    Chennai  01-01-2008   
2          IND         19  Bangalore  Bangalore  01-11-2012   
3          IND         10  New Delhi    Gurgaon  01-06-2012   
4          IND         19  Bangalore  Bangalore  01-01-2001   

                    company_permalink  \
0                 /ORGANIZATION/-FAME   
1             /ORGANIZATION/1000LOOKZ   
2                /ORGANIZATION/1CLICK   
3      /ORGANIZATION/21DIAMONDS-INDIA   
4  /ORGANIZATION/247-LEARNING-PRIVATE   

                           funding_round_permalink funding_round_type  \
0  /funding-round/9a01d05418af9f794eebff7ace91f638            venture   
1  /funding-round/e07f6e580046d00c7aeb33fafb9d90a3            venture   
2  /funding-round/19c77bc4b09e68e62f9a376d24115be4               seed   
3  /funding-round/6de7ffef8091ba9f33821f4b861f434a            venture   
4  /funding-round/4cd5cd165bc869bc40705a9e4b626bee            venture   

  funding_round_code   funded_at  raised_amount_usd Country Code  
0                  B  05-01-2015        10000000.00   India  IND  
1                NaN  22-07-2013                nan   India  IND  
2                NaN  26-01-2014                nan   India  IND  
3                  C  15-11-2012         6369507.00   India  IND  
4                NaN  06-11-2007         4000000.00   India  IND

In [44]:
master_frame.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 86441 entries, 0 to 86440
Data columns (total 18 columns):
permalink                  86441 non-null object
name                       86441 non-null object
homepage_url               82355 non-null object
category_list              85170 non-null object
status                     86441 non-null object
country_code               86441 non-null object
state_code                 85323 non-null object
region                     85879 non-null object
city                       85881 non-null object
founded_at                 73647 non-null object
company_permalink          86441 non-null object
funding_round_permalink    86441 non-null object
funding_round_type         86441 non-null object
funding_round_code         23838 non-null object
funded_at                  86441 non-null object
raised_amount_usd          73541 non-null float64
Country                    86441 non-null object
Code                       86441 non-null object
dtypes: 

- ### Find the Country where most investment is done in funding type = Most suitable Funding type (Venture)

In [45]:
grpby_obj = master_frame.loc[master_frame.funding_round_type == most_suitable_ft ].groupby('Country')['raised_amount_usd']

In [46]:
top9 = pd.DataFrame(grpby_obj.sum().sort_values(ascending = False)).iloc[0:9,]

In [47]:
top9

raised_amount_usd
Country                          
United States     422510842796.00
United Kingdom     20245627416.00
India              14391858718.00
Canada              9583332317.00
Singapore           2793917856.00
Ireland             1676131350.00
Australia           1322934948.00
New Zealand          448316383.00
Nigeria              245440000.00

In [48]:
## Top three country
top9.head(3)

raised_amount_usd
Country                          
United States     422510842796.00
United Kingdom     20245627416.00
India              14391858718.00

- ### Checkpoint 4 Sector Anlaysis 1
- ### Reading `mappings` file

In [49]:
mappings = pd.read_csv(path_initial + 'mapping.csv')
mappings

category_list  Automotive & Sports  Blanks  \
0                           NaN                    0       1   
1                            3D                    0       0   
2                   3D Printing                    0       0   
3                 3D Technology                    0       0   
4                    Accounting                    0       0   
5              Active Lifestyle                    0       0   
6                  Ad Targeting                    0       0   
7            Advanced Materials                    0       0   
8              Adventure Travel                    1       0   
9                   Advertising                    0       0   
10        Advertising Exchanges                    0       0   
11         Advertising Networks                    0       0   
12        Advertising Platforms                    0       0   
13                       Advice                    0       0   
14                    Aerospace                    1       0   
15                  Agriculture                    0       0   
16        Air Pollution Control                    0       0   
17                   Algorithms                    0       0   
18                  All Markets                    0       0   
19                 All Students                    0       0   
20          Alter0tive Medicine                    0       0   
21                       Alumni                    0       0   
22                     A0lytics                    0       0   
23                      Android                    0       0   
24                       Angels                    0       0   
25                  Animal Feed                    0       0   
26   Anything Capital Intensive                    0       0   
27                App Discovery                    0       0   
28                App Marketing                    0       0   
29                   App Stores                    0       0   
..                          ...                  ...     ...   
658                   Usability                    0       0   
659      User Experience Design                    0       0   
660              User Interface                    0       0   
661                   Utilities                    0       0   
662     Vending and Concessions                    0       0   
663             Venture Capital                    0       0   
664                   Veteri0ry                    0       0   
665                       Video                    0       0   
666          Video Conferencing                    0       0   
667                 Video Games                    0       0   
668             Video on Demand                    0       0   
669             Video Streaming                    0       0   
670          Virtual Workforces                    0       0   
671                        VoIP                    0       0   
672             Waste Ma0gement                    0       0   
673                       Watch                    0       0   
674                       Water                    0       0   
675          Water Purification                    0       0   
676                   Wearables                    0       0   
677                Web Browsers                    0       0   
678                  Web Design                    0       0   
679             Web Development                    0       0   
680                 Web Hosting                    0       0   
681                   Web Tools                    0       0   
682                    Weddings                    0       0   
683                   Wholesale                    0       0   
684            Wine And Spirits                    0       0   
685                    Wireless                    0       0   
686                       Women                    0       0   
687                Young Adults                    0       0   

     Cleantech / Semiconductors  Entertainment  H

In [50]:
#Lets check mapping file

mappings.shape

(688, 10)

In [51]:
mappings.describe()

Automotive & Sports  Blanks  Cleantech / Semiconductors  Entertainment  \
count               688.00  688.00                      688.00         688.00   
mean                  0.03    0.00                        0.08           0.13   
std                   0.18    0.04                        0.27           0.34   
min                   0.00    0.00                        0.00           0.00   
25%                   0.00    0.00                        0.00           0.00   
50%                   0.00    0.00                        0.00           0.00   
75%                   0.00    0.00                        0.00           0.00   
max                   1.00    1.00                        1.00           1.00   

       Health  Manufacturing  News, Search and Messaging  Others  \
count  688.00         688.00                      688.00  688.00   
mean     0.09           0.06                        0.10    0.28   
std      0.29           0.23                        0.31    0.45   
min      0.00           0.00                        0.00    0.00   
25%      0.00           0.00                        0.00    0.00   
50%      0.00           0.00                        0.00    0.00   
75%      0.00           0.00                        0.00    1.00   
max      1.00           1.00                        1.00    1.00   

       Social, Finance, Analytics, Advertising  
count                                   688.00  
mean                                      0.22  
std                                       0.42  
min                                       0.00  
25%                                       0.00  
50%                                       0.00  
75%                                       0.00  
max                                       1.00

In [52]:
mappings.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 688 entries, 0 to 687
Data columns (total 10 columns):
category_list                              687 non-null object
Automotive & Sports                        688 non-null int64
Blanks                                     688 non-null int64
Cleantech / Semiconductors                 688 non-null int64
Entertainment                              688 non-null int64
Health                                     688 non-null int64
Manufacturing                              688 non-null int64
News, Search and Messaging                 688 non-null int64
Others                                     688 non-null int64
Social, Finance, Analytics, Advertising    688 non-null int64
dtypes: int64(9), object(1)
memory usage: 53.8+ KB


In [53]:
mappings.loc[mappings.Blanks == 1]

category_list  Automotive & Sports  Blanks  Cleantech / Semiconductors  \
0           NaN                    0       1                           0   

   Entertainment  Health  Manufacturing  News, Search and Messaging  Others  \
0              0       0              0                           0       0   

   Social, Finance, Analytics, Advertising  
0                                        0

- Lets Validate if any category is classified in more than one or none category

In [54]:
mappings.loc[mappings.sum(axis=1) != 1]

Empty DataFrame
Columns: [category_list, Automotive & Sports, Blanks, Cleantech / Semiconductors, Entertainment, Health, Manufacturing, News, Search and Messaging, Others, Social, Finance, Analytics, Advertising]
Index: []

In [55]:
np.sum(mappings.category_list.duplicated() == False)

688

In [56]:
master_frame['primary_sector'] , master_frame['temp_sector'] = master_frame['category_list'] .str.split('|',1).str

master_frame.head()  

permalink           name  \
0                 /ORGANIZATION/-FAME          #fame   
1             /ORGANIZATION/1000LOOKZ      1000Lookz   
2                /ORGANIZATION/1CLICK         1CLICK   
3      /ORGANIZATION/21DIAMONDS-INDIA     21Diamonds   
4  /ORGANIZATION/247-LEARNING-PRIVATE  24x7 Learning   

                  homepage_url             category_list     status  \
0           http://livfame.com                     Media  operating   
1         http://1000lookz.com                    Beauty  operating   
2             http://1click.io               Chat|Mobile   acquired   
3     http://www.21diamonds.de                E-Commerce  operating   
4  http://www.24x7learning.com  EdTech|Education|Systems  operating   

  country_code state_code     region       city  founded_at  \
0          IND         16     Mumbai     Mumbai         NaN   
1          IND         25    Chennai    Chennai  01-01-2008   
2          IND         19  Bangalore  Bangalore  01-11-2012   
3          IND         10  New Delhi    Gurgaon  01-06-2012   
4          IND         19  Bangalore  Bangalore  01-01-2001   

                    company_permalink  \
0                 /ORGANIZATION/-FAME   
1             /ORGANIZATION/1000LOOKZ   
2                /ORGANIZATION/1CLICK   
3      /ORGANIZATION/21DIAMONDS-INDIA   
4  /ORGANIZATION/247-LEARNING-PRIVATE   

                           funding_round_permalink funding_round_type  \
0  /funding-round/9a01d05418af9f794eebff7ace91f638            venture   
1  /funding-round/e07f6e580046d00c7aeb33fafb9d90a3            venture   
2  /funding-round/19c77bc4b09e68e62f9a376d24115be4               seed   
3  /funding-round/6de7ffef8091ba9f33821f4b861f434a            venture   
4  /funding-round/4cd5cd165bc869bc40705a9e4b626bee            venture   

  funding_round_code   funded_at  raised_amount_usd Country Code  \
0                  B  05-01-2015        10000000.00   India  IND   
1                NaN  22-07-2013                nan   India  IND   
2                NaN  26-01-2014                nan   India  IND   
3                  C  15-11-2012         6369507.00   India  IND   
4                NaN  06-11-2007         4000000.00   India  IND   

  primary_sector        temp_sector  
0          Media                NaN  
1         Beauty                NaN  
2           Chat             Mobile  
3     E-Commerce                NaN  
4         EdTech  Education|Systems

In [57]:
master_frame=master_frame.drop("temp_sector",axis= 1) #dropping temporary column

master_frame.head()

permalink           name  \
0                 /ORGANIZATION/-FAME          #fame   
1             /ORGANIZATION/1000LOOKZ      1000Lookz   
2                /ORGANIZATION/1CLICK         1CLICK   
3      /ORGANIZATION/21DIAMONDS-INDIA     21Diamonds   
4  /ORGANIZATION/247-LEARNING-PRIVATE  24x7 Learning   

                  homepage_url             category_list     status  \
0           http://livfame.com                     Media  operating   
1         http://1000lookz.com                    Beauty  operating   
2             http://1click.io               Chat|Mobile   acquired   
3     http://www.21diamonds.de                E-Commerce  operating   
4  http://www.24x7learning.com  EdTech|Education|Systems  operating   

  country_code state_code     region       city  founded_at  \
0          IND         16     Mumbai     Mumbai         NaN   
1          IND         25    Chennai    Chennai  01-01-2008   
2          IND         19  Bangalore  Bangalore  01-11-2012   
3          IND         10  New Delhi    Gurgaon  01-06-2012   
4          IND         19  Bangalore  Bangalore  01-01-2001   

                    company_permalink  \
0                 /ORGANIZATION/-FAME   
1             /ORGANIZATION/1000LOOKZ   
2                /ORGANIZATION/1CLICK   
3      /ORGANIZATION/21DIAMONDS-INDIA   
4  /ORGANIZATION/247-LEARNING-PRIVATE   

                           funding_round_permalink funding_round_type  \
0  /funding-round/9a01d05418af9f794eebff7ace91f638            venture   
1  /funding-round/e07f6e580046d00c7aeb33fafb9d90a3            venture   
2  /funding-round/19c77bc4b09e68e62f9a376d24115be4               seed   
3  /funding-round/6de7ffef8091ba9f33821f4b861f434a            venture   
4  /funding-round/4cd5cd165bc869bc40705a9e4b626bee            venture   

  funding_round_code   funded_at  raised_amount_usd Country Code  \
0                  B  05-01-2015        10000000.00   India  IND   
1                NaN  22-07-2013                nan   India  IND   
2                NaN  26-01-2014                nan   India  IND   
3                  C  15-11-2012         6369507.00   India  IND   
4                NaN  06-11-2007         4000000.00   India  IND   

  primary_sector  
0          Media  
1         Beauty  
2           Chat  
3     E-Commerce  
4         EdTech

In [58]:
# Converting values in Mapping Dataframe to Column Headings to get Sector names for each Category List
temp_mapdf = pd.melt(mappings , id_vars = ['category_list'] , var_name = 'main_sector')
mapdf = temp_mapdf[temp_mapdf['value']>0]
mapdf.head()

category_list          main_sector  value
8   Adventure Travel  Automotive & Sports      1
14         Aerospace  Automotive & Sports      1
45              Auto  Automotive & Sports      1
46   Automated Kiosk  Automotive & Sports      1
47        Automotive  Automotive & Sports      1

In [59]:
#   So lets check the unique value of primary sector in master data frame 
pd.unique(master_frame.primary_sector)


array(['Media', 'Beauty', 'Chat', 'E-Commerce', 'EdTech', 'Internet',
       '3D', 'Mobile', 'Games', 'Software', 'Finance', 'Consulting',
       'Advertising', 'Business Services', 'Crowdsourcing', 'Energy',
       'Curated Web', 'Internet Service Providers', 'Cloud Management',
       'Child Care', 'Classifieds', 'Entertainment',
       'Advertising Platforms', 'Analytics', 'Tracking', 'Match-Making',
       'Consumer Internet', 'Health and Wellness', 'Aerospace',
       'BPO Services', 'Hardware + Software', 'Apps', 'Electric Vehicles',
       'Automotive', 'Networking', 'Digital Media', 'Enterprise Software',
       'Content', 'Credit', 'Education', 'Information Technology',
       'Human Resources', 'Concentrated Solar Power', 'Online Scheduling',
       'Real Estate', 'Consumer Electronics', 'Network Security',
       'Journalism', 'Clean Technology', 'Cloud Security', 'Guides',
       'Security', 'Collaboration', 'Consumer Goods', 'Biotechnology',
       'SaaS', 'Financial Servi

In [60]:
pd.unique(mappings.category_list)

array([nan, '3D', '3D Printing', '3D Technology', 'Accounting',
       'Active Lifestyle', 'Ad Targeting', 'Advanced Materials',
       'Adventure Travel', 'Advertising', 'Advertising Exchanges',
       'Advertising Networks', 'Advertising Platforms', 'Advice',
       'Aerospace', 'Agriculture', 'Air Pollution Control', 'Algorithms',
       'All Markets', 'All Students', 'Alter0tive Medicine', 'Alumni',
       'A0lytics', 'Android', 'Angels', 'Animal Feed',
       'Anything Capital Intensive', 'App Discovery', 'App Marketing',
       'App Stores', 'Application Performance Monitoring',
       'Application Platforms', 'Apps', 'Aquaculture', 'Architecture',
       'Archiving', 'Art', 'Artificial Intelligence', 'Artists Globally',
       'Assisitive Technology', 'Assisted Living', 'Auctions', 'Audio',
       'Audiobooks', 'Augmented Reality', 'Auto', 'Automated Kiosk',
       'Automotive', 'B2B', 'B2B Express Delivery', 'Babies',
       'Baby Accessories', 'Baby Boomers', 'Baby Safety', 'B

- ### We can see there are some category where 0 need to be replaced by na

In [61]:
mappings['category_list'] = mappings['category_list'].str.replace('0','na')
pd.unique(mappings.category_list)

array([nan, '3D', '3D Printing', '3D Technology', 'Accounting',
       'Active Lifestyle', 'Ad Targeting', 'Advanced Materials',
       'Adventure Travel', 'Advertising', 'Advertising Exchanges',
       'Advertising Networks', 'Advertising Platforms', 'Advice',
       'Aerospace', 'Agriculture', 'Air Pollution Control', 'Algorithms',
       'All Markets', 'All Students', 'Alternative Medicine', 'Alumni',
       'Analytics', 'Android', 'Angels', 'Animal Feed',
       'Anything Capital Intensive', 'App Discovery', 'App Marketing',
       'App Stores', 'Application Performance Monitoring',
       'Application Platforms', 'Apps', 'Aquaculture', 'Architecture',
       'Archiving', 'Art', 'Artificial Intelligence', 'Artists Globally',
       'Assisitive Technology', 'Assisted Living', 'Auctions', 'Audio',
       'Audiobooks', 'Augmented Reality', 'Auto', 'Automated Kiosk',
       'Automotive', 'B2B', 'B2B Express Delivery', 'Babies',
       'Baby Accessories', 'Baby Boomers', 'Baby Safety', 

In [62]:
mappings['category_list'] = mappings['category_list'].str.replace('.na','.0')
pd.unique(mappings.category_list)

array([nan, '3D', '3D Printing', '3D Technology', 'Accounting',
       'Active Lifestyle', 'Ad Targeting', 'Advanced Materials',
       'Adventure Travel', 'Advertising', 'Advertising Exchanges',
       'Advertising Networks', 'Advertising Platforms', 'Advice',
       'Aerospace', 'Agriculture', 'Air Pollution Control', 'Algorithms',
       'All Markets', 'All Students', 'Alte.0tive Medicine', 'Alumni',
       '.0lytics', 'Android', 'Angels', 'Animal Feed',
       'Anything Capital Intensive', 'App Discovery', 'App Marketing',
       'App Stores', 'Application Performance Monitoring',
       'Application Platforms', 'Apps', 'Aquaculture', 'Architecture',
       'Archiving', 'Art', 'Artificial Intelligence', 'Artists Globally',
       'Assisitive Technology', 'Assisted Living', 'Auctions', 'Audio',
       'Audiobooks', 'Augmented Reality', 'Auto', 'Automated Kiosk',
       'Automotive', 'B2B', 'B2B Express Delivery', 'Babies',
       'Baby Accessories', 'Baby Boomers', 'Baby Safety', 'B

- ### Now inner join data based upon category list and primary sector , As we are doing inner join so rows with all the blank category list will be nullified

In [63]:
master_frame =pd.merge(master_frame ,mapdf[['category_list','main_sector']], how='inner', left_on=['primary_sector'], right_on=['category_list'])
master_frame.head()

permalink              name  \
0                /ORGANIZATION/-FAME             #fame   
1                /ORGANIZATION/INC42             Inc42   
2    /ORGANIZATION/MANAS-INFORMATICS  Manas Informatic   
3  /ORGANIZATION/SUN-ANIMATICS-INDIA     Sun Animatics   
4      /ORGANIZATION/A-B-PRODUCTIONS   A.B Productions   

                  homepage_url               category_list_x     status  \
0           http://livfame.com                         Media  operating   
1            http://inc42.com/                         Media  operating   
2        http://www.topnews.in                    Media|News  operating   
3  http://www.sunanimatics.com  Media|News|Photography|Video  operating   
4                          NaN              Media|Music|News  operating   

  country_code state_code       region         city  founded_at  \
0          IND         16       Mumbai       Mumbai         NaN   
1          IND          7    New Delhi    New Delhi  01-01-2014   
2          IND         23       Mohali       Mohali  05-05-2007   
3          IND          2    Hyderabad    Hyderabad  16-06-2014   
4          USA         CA  Los Angeles  Los Angeles  11-12-2012   

       ...                                funding_round_permalink  \
0      ...        /funding-round/9a01d05418af9f794eebff7ace91f638   
1      ...        /funding-round/d5e89dc6cd96d3a786a9e19cc4f8a907   
2      ...        /funding-round/719e50301803d3918ffa558fc877e41c   
3      ...        /funding-round/d1510e295203f7ff9d5afd270930bec1   
4      ...        /funding-round/23bab180ab73f0dcc3f555bebdf89ade   

    funding_round_type funding_round_code   funded_at raised_amount_usd  \
0              venture                  B  05-01-2015       10000000.00   
1                 seed                NaN  08-01-2014               nan   
2              venture                NaN  02-02-2008         160000.00   
3  equity_crowdfunding                NaN  13-11-2010               nan   
4       debt_financing                NaN  21-01-2014               nan   

         Country Code primary_sector category_list_y    main_sector  
0          India  IND          Media           Media  Entertainment  
1          India  IND          Media           Media  Entertainment  
2          India  IND          Media           Media  Entertainment  
3          India  IND          Media           Media  Entertainment  
4  United States  USA          Media           Media  Entertainment  

[5 rows x 21 columns]

In [64]:
#Renaming the column category_list_x to category_list 
master_frame.rename(columns={'category_list_x':'category_list'}, inplace = 'True')
master_frame.head()

permalink              name  \
0                /ORGANIZATION/-FAME             #fame   
1                /ORGANIZATION/INC42             Inc42   
2    /ORGANIZATION/MANAS-INFORMATICS  Manas Informatic   
3  /ORGANIZATION/SUN-ANIMATICS-INDIA     Sun Animatics   
4      /ORGANIZATION/A-B-PRODUCTIONS   A.B Productions   

                  homepage_url                 category_list     status  \
0           http://livfame.com                         Media  operating   
1            http://inc42.com/                         Media  operating   
2        http://www.topnews.in                    Media|News  operating   
3  http://www.sunanimatics.com  Media|News|Photography|Video  operating   
4                          NaN              Media|Music|News  operating   

  country_code state_code       region         city  founded_at  \
0          IND         16       Mumbai       Mumbai         NaN   
1          IND          7    New Delhi    New Delhi  01-01-2014   
2          IND         23       Mohali       Mohali  05-05-2007   
3          IND          2    Hyderabad    Hyderabad  16-06-2014   
4          USA         CA  Los Angeles  Los Angeles  11-12-2012   

       ...                                funding_round_permalink  \
0      ...        /funding-round/9a01d05418af9f794eebff7ace91f638   
1      ...        /funding-round/d5e89dc6cd96d3a786a9e19cc4f8a907   
2      ...        /funding-round/719e50301803d3918ffa558fc877e41c   
3      ...        /funding-round/d1510e295203f7ff9d5afd270930bec1   
4      ...        /funding-round/23bab180ab73f0dcc3f555bebdf89ade   

    funding_round_type funding_round_code   funded_at raised_amount_usd  \
0              venture                  B  05-01-2015       10000000.00   
1                 seed                NaN  08-01-2014               nan   
2              venture                NaN  02-02-2008         160000.00   
3  equity_crowdfunding                NaN  13-11-2010               nan   
4       debt_financing                NaN  21-01-2014               nan   

         Country Code primary_sector category_list_y    main_sector  
0          India  IND          Media           Media  Entertainment  
1          India  IND          Media           Media  Entertainment  
2          India  IND          Media           Media  Entertainment  
3          India  IND          Media           Media  Entertainment  
4  United States  USA          Media           Media  Entertainment  

[5 rows x 21 columns]

In [65]:
#dropping column category_list_y
master_frame = master_frame.drop(['category_list_y'] , axis = 1)
master_frame.head()

permalink              name  \
0                /ORGANIZATION/-FAME             #fame   
1                /ORGANIZATION/INC42             Inc42   
2    /ORGANIZATION/MANAS-INFORMATICS  Manas Informatic   
3  /ORGANIZATION/SUN-ANIMATICS-INDIA     Sun Animatics   
4      /ORGANIZATION/A-B-PRODUCTIONS   A.B Productions   

                  homepage_url                 category_list     status  \
0           http://livfame.com                         Media  operating   
1            http://inc42.com/                         Media  operating   
2        http://www.topnews.in                    Media|News  operating   
3  http://www.sunanimatics.com  Media|News|Photography|Video  operating   
4                          NaN              Media|Music|News  operating   

  country_code state_code       region         city  founded_at  \
0          IND         16       Mumbai       Mumbai         NaN   
1          IND          7    New Delhi    New Delhi  01-01-2014   
2          IND         23       Mohali       Mohali  05-05-2007   
3          IND          2    Hyderabad    Hyderabad  16-06-2014   
4          USA         CA  Los Angeles  Los Angeles  11-12-2012   

                   company_permalink  \
0                /ORGANIZATION/-FAME   
1                /ORGANIZATION/INC42   
2    /ORGANIZATION/MANAS-INFORMATICS   
3  /ORGANIZATION/SUN-ANIMATICS-INDIA   
4      /ORGANIZATION/A-B-PRODUCTIONS   

                           funding_round_permalink   funding_round_type  \
0  /funding-round/9a01d05418af9f794eebff7ace91f638              venture   
1  /funding-round/d5e89dc6cd96d3a786a9e19cc4f8a907                 seed   
2  /funding-round/719e50301803d3918ffa558fc877e41c              venture   
3  /funding-round/d1510e295203f7ff9d5afd270930bec1  equity_crowdfunding   
4  /funding-round/23bab180ab73f0dcc3f555bebdf89ade       debt_financing   

  funding_round_code   funded_at  raised_amount_usd        Country Code  \
0                  B  05-01-2015        10000000.00          India  IND   
1                NaN  08-01-2014                nan          India  IND   
2                NaN  02-02-2008          160000.00          India  IND   
3                NaN  13-11-2010                nan          India  IND   
4                NaN  21-01-2014                nan  United States  USA   

  primary_sector    main_sector  
0          Media  Entertainment  
1          Media  Entertainment  
2          Media  Entertainment  
3          Media  Entertainment  
4          Media  Entertainment

In [66]:
master_frame.shape

(79959, 20)

-  ### Checkpoint 5- Sector Analysis 2

- ### As Spark Funds want to invest only between 5 -15 Million so we applied check on raised amount funded for 5 to 15 million

In [67]:
## First data frame D1 for top most Country for funded type = 'Venture'

D1=pd.DataFrame(master_frame.loc[(master_frame['Country'] ==top9.index[0]) & (master_frame['funding_round_type']== most_suitable_ft)& 
                                 (master_frame['raised_amount_usd'].between(5000000 , 15000000))])

D1['total_amount_invested']= D1.groupby('main_sector')['raised_amount_usd'].transform('sum')
D1['total_investment_count']= D1.groupby('main_sector')['raised_amount_usd'].transform('count')

D1.head()





permalink                 name  \
9   /ORGANIZATION/ALL-DEF-DIGITAL      All Def Digital   
33       /ORGANIZATION/CHEFS-FEED            ChefsFeed   
69   /ORGANIZATION/HUFFINGTONPOST  The Huffington Post   
70   /ORGANIZATION/HUFFINGTONPOST  The Huffington Post   
87        /ORGANIZATION/MATCHMINE            MatchMine   

                     homepage_url                           category_list  \
9        http://alldefdigital.com                                   Media   
33       http://www.chefsfeed.com     Media|Mobile|Restaurants|Technology   
69  http://www.huffingtonpost.com                   Media|News|Publishing   
70  http://www.huffingtonpost.com                   Media|News|Publishing   
87           http://matchmine.com  Media|News|Reviews and Recommendations   

       status country_code state_code         region           city  \
9   operating          USA         CA    Los Angeles    Los Angeles   
33  operating          USA         CA    SF Bay Area  San Francisco   
69   acquired          USA         NY  New York City       New York   
70   acquired          USA         NY  New York City       New York   
87     closed          USA         MA         Boston        Needham   

    founded_at          ...           funding_round_type funding_round_code  \
9          NaN          ...                      venture                  A   
33  01-01-2012          ...                      venture                  A   
69  09-05-2005          ...                      venture                  A   
70  09-05-2005          ...                      venture                  B   
87  01-01-2007          ...                      venture                  A   

     funded_at raised_amount_usd        Country  Code primary_sector  \
9   06-08-2014        5000000.00  United States   USA          Media   
33  26-02-2015        5000000.00  United States   USA          Media   
69  01-08-2006        5000000.00  United States   USA          Media   
70  01-09-2007        5000000.00  United States   USA          Media   
87  01-09-2007       10000000.00  United States   USA          Media   

      main_sector total_amount_invested total_investment_count  
9   Entertainment         5000212982.00                    580  
33  Entertainment         5000212982.00                    580  
69  Entertainment         5000212982.00                    580  
70  Entertainment         5000212982.00                    580  
87  Entertainment         5000212982.00                    580  

[5 rows x 22 columns]

In [68]:
D1.tail()

permalink                   name  \
79861                 /ORGANIZATION/RXWIKI           RxWiki, Inc.   
79862    /ORGANIZATION/TIKVAH-THERAPEUTICS    Tikvah Therapeutics   
79863  /ORGANIZATION/SAHARA-MEDIA-HOLDINGS  Sahara Media Holdings   
79902               /ORGANIZATION/THISNEXT               ThisNext   
79917         /ORGANIZATION/USER-INTERFACE         User Interface   

                       homepage_url  \
79861         http://www.rxwiki.com   
79862                           NaN   
79863                           NaN   
79902       http://www.thisnext.com   
79917  http://www.userinterface.com   

                                           category_list     status  \
79861  Independent Pharmacies|Mobile Health|Pharmaceu...  operating   
79862                Independent Pharmacies|Neuroscience  operating   
79863              Lifestyle Businesses|Media|Publishing  operating   
79902                         Social Buying|Social Media  operating   
79917                                     User Interface  operating   

      country_code state_code         region             city  founded_at  \
79861          USA         TX         Austin           Austin  01-03-2012   
79862          USA         GA        Atlanta          Atlanta         NaN   
79863          USA         NY  New York City         New York         NaN   
79902          USA         CA    Los Angeles     Santa Monica  01-12-2005   
79917          USA         CA    Bakersfield  California City         NaN   

               ...           funding_round_type funding_round_code  \
79861          ...                      venture                  A   
79862          ...                      venture                NaN   
79863          ...                      venture                NaN   
79902          ...                      venture                  B   
79917          ...                      venture                  A   

        funded_at raised_amount_usd        Country  Code  \
79861  16-10-2015        5745869.00  United States   USA   
79862  14-07-2007       10000000.00  United States   USA   
79863  01-10-2008        8157678.00  United States   USA   
79902  01-01-2008        5000000.00  United States   USA   
79917  24-01-2001        8250000.00  United States   USA   

               primary_sector                              main_sector  \
79861  Independent Pharmacies                                   Health   
79862  Independent Pharmacies                                   Health   
79863    Lifestyle Businesses                            Entertainment   
79902           Social Buying  Social, Finance, Analytics, Advertising   
79917          User Interface                                   Others   

      total_amount_invested total_investment_count  
79861         8200859357.00                    908  
79862         8200859357.00                    908  
79863         5000212982.00                    580  
79902        16680695048.00                   1912  
79917        26089280296.00                   2923  

[5 rows x 22 columns]

In [69]:

## Second data frame D2 for Second  Country for funded type = 'Venture'

D2=pd.DataFrame(master_frame.loc[(master_frame['Country'] ==top9.index[1]) & (master_frame['funding_round_type']== most_suitable_ft)& 
                                 (master_frame['raised_amount_usd'].between(5000000 , 15000000))])

D2['total_amount_invested']= D2.groupby('main_sector')['raised_amount_usd'].transform('sum')
D2['total_investment_count']= D2.groupby('main_sector')['raised_amount_usd'].transform('count')

D2.head()




permalink          name            homepage_url  \
194          /ORGANIZATION/90MIN         90min    http://www.90min.com   
195          /ORGANIZATION/90MIN         90min    http://www.90min.com   
397        /ORGANIZATION/WAHANDA       Wahanda  http://www.wahanda.com   
3057  /ORGANIZATION/ACS-CLOTHING  ACS Clothing   http://hirewear.co.uk   
3058  /ORGANIZATION/ACS-CLOTHING  ACS Clothing   http://hirewear.co.uk   

                                          category_list     status  \
194                 Media|News|Publishing|Soccer|Sports  operating   
195                 Media|News|Publishing|Soccer|Sports  operating   
397   Beauty|Curated Web|Fitness|Health and Wellness...   acquired   
3057                                         E-Commerce  operating   
3058                                         E-Commerce  operating   

     country_code state_code   region     city  founded_at  \
194           GBR         H9   London   London  01-01-2011   
195           GBR         H9   London   London  01-01-2011   
397           GBR         H9   London   London  14-02-2008   
3057          GBR         V2  Glasgow  Glasgow  01-01-1997   
3058          GBR         V2  Glasgow  Glasgow  01-01-1997   

              ...           funding_round_type funding_round_code   funded_at  \
194           ...                      venture                NaN  06-10-2015   
195           ...                      venture                NaN  07-05-2013   
397           ...                      venture                  B  17-10-2011   
3057          ...                      venture                NaN  28-09-2015   
3058          ...                      venture                NaN  02-03-2014   

     raised_amount_usd         Country  Code primary_sector    main_sector  \
194        15000000.00  United Kingdom   GBR          Media  Entertainment   
195         5800000.00  United Kingdom   GBR          Media  Entertainment   
397         8682611.00  United Kingdom   GBR         Beauty         Health   
3057       13359377.00  United Kingdom   GBR     E-Commerce         Others   
3058       12879637.00  United Kingdom   GBR     E-Commerce         Others   

     total_amount_invested total_investment_count  
194           461511059.00                     54  
195           461511059.00                     54  
397           214537510.00                     24  
3057         1249124289.00                    143  
3058         1249124289.00                    143  

[5 rows x 22 columns]

In [70]:

## Third data frame D3 for Third  Country for funded type = 'Venture'

D3=pd.DataFrame(master_frame.loc[(master_frame['Country'] ==top9.index[2]) & (master_frame['funding_round_type']== most_suitable_ft)& 
                                 (master_frame['raised_amount_usd'].between(5000000 , 15000000))])

D3['total_amount_invested']= D3.groupby('main_sector')['raised_amount_usd'].transform('sum')
D3['total_investment_count']= D3.groupby('main_sector')['raised_amount_usd'].transform('count')

D3.head()



permalink           name              homepage_url  \
0               /ORGANIZATION/-FAME          #fame        http://livfame.com   
547  /ORGANIZATION/21DIAMONDS-INDIA     21Diamonds  http://www.21diamonds.de   
558     /ORGANIZATION/BESTYLISH-COM  beStylish.com  http://www.bestylish.com   
585         /ORGANIZATION/CARATLANE      CaratLane  http://www.caratlane.com   
586         /ORGANIZATION/CARATLANE      CaratLane  http://www.caratlane.com   

                category_list     status country_code state_code     region  \
0                       Media  operating          IND         16     Mumbai   
547                E-Commerce  operating          IND         10  New Delhi   
558  E-Commerce|Fashion|Shoes     closed          IND         10  New Delhi   
585        E-Commerce|Jewelry  operating          IND         25    Chennai   
586        E-Commerce|Jewelry  operating          IND         25    Chennai   

        city  founded_at          ...           funding_round_type  \
0     Mumbai         NaN          ...                      venture   
547  Gurgaon  01-06-2012          ...                      venture   
558  Gurgaon  10-05-2011          ...                      venture   
585  Chennai  01-10-2008          ...                      venture   
586  Chennai  01-10-2008          ...                      venture   

    funding_round_code   funded_at raised_amount_usd Country  Code  \
0                    B  05-01-2015       10000000.00   India   IND   
547                  C  15-11-2012        6369507.00   India   IND   
558                  A  01-06-2012       10000000.00   India   IND   
585                  A  22-06-2011        6000000.00   India   IND   
586                  C  13-05-2013       15000000.00   India   IND   

    primary_sector    main_sector total_amount_invested total_investment_count  
0            Media  Entertainment          280830000.00                     33  
547     E-Commerce         Others         1007409507.00                    109  
558     E-Commerce         Others         1007409507.00                    109  
585     E-Commerce         Others         1007409507.00                    109  
586     E-Commerce         Others         1007409507.00                    109  

[5 rows x 22 columns]

- ### Check the count of investment for Top 3 Countries

In [71]:
Country1_count_investment = D1['raised_amount_usd'].count()
print(Country1_count_investment)

Country2_count_investment = D2['raised_amount_usd'].count()
print(Country2_count_investment)

Country3_count_investment = D3['raised_amount_usd'].count()
print(Country3_count_investment)


11235
584
301


In [72]:
D2.count()

permalink                  584
name                       584
homepage_url               555
category_list              577
status                     584
country_code               584
state_code                 569
region                     554
city                       554
founded_at                 462
company_permalink          584
funding_round_permalink    584
funding_round_type         584
funding_round_code         360
funded_at                  584
raised_amount_usd          584
Country                    584
Code                       584
primary_sector             577
main_sector                584
total_amount_invested      584
total_investment_count     584
dtype: int64

- ### Check the total amount of investment for each country

In [73]:
Country1_amount_investment = D1['raised_amount_usd'].sum()
print(Country1_amount_investment)

Country2_amount_investment = D2['raised_amount_usd'].sum()
print(Country2_amount_investment)

Country3_amount_investment = D3['raised_amount_usd'].sum()
print(Country3_amount_investment)


100426287841.0
5086469206.0
2710537552.0


- ### Check the top 3 sector (No of investment wise)

In [74]:
D11=D1.sort_values(by=['total_investment_count'] ,ascending = False )


D11=D11.drop_duplicates('total_investment_count',keep='first')


D11top=D11.head(3)

D11top[['main_sector','total_investment_count']]



main_sector  total_investment_count
38313                                   Others                    2923
41540               Cleantech / Semiconductors                    2297
16974  Social, Finance, Analytics, Advertising                    1912

In [75]:
D21=D2.sort_values(by=['total_investment_count'] ,ascending = False )


D21=D21.drop_duplicates('total_investment_count',keep='first')


D21top=D21.head(3)

D21top[['main_sector','total_investment_count']]

main_sector  total_investment_count
13838                                   Others                     143
37735               Cleantech / Semiconductors                     127
49582  Social, Finance, Analytics, Advertising                      98

In [76]:
D31=D3.sort_values(by=['total_investment_count'] ,ascending = False )


D31=D31.drop_duplicates('total_investment_count',keep='first')


D31top=D31.head(3)

D31top[['main_sector','total_investment_count']]

main_sector  total_investment_count
32716                      Others                     109
27033  News, Search and Messaging                      52
51252               Entertainment                      33

- ### find out which company received the highest investment for Sector(Point 3) 

In [77]:
D1A=D1
D1A['company_amount_invested']= D1A.groupby(['main_sector', 'permalink'])['raised_amount_usd'].transform('sum')

In [78]:
D1A.head()

permalink                 name  \
9   /ORGANIZATION/ALL-DEF-DIGITAL      All Def Digital   
33       /ORGANIZATION/CHEFS-FEED            ChefsFeed   
69   /ORGANIZATION/HUFFINGTONPOST  The Huffington Post   
70   /ORGANIZATION/HUFFINGTONPOST  The Huffington Post   
87        /ORGANIZATION/MATCHMINE            MatchMine   

                     homepage_url                           category_list  \
9        http://alldefdigital.com                                   Media   
33       http://www.chefsfeed.com     Media|Mobile|Restaurants|Technology   
69  http://www.huffingtonpost.com                   Media|News|Publishing   
70  http://www.huffingtonpost.com                   Media|News|Publishing   
87           http://matchmine.com  Media|News|Reviews and Recommendations   

       status country_code state_code         region           city  \
9   operating          USA         CA    Los Angeles    Los Angeles   
33  operating          USA         CA    SF Bay Area  San Francisco   
69   acquired          USA         NY  New York City       New York   
70   acquired          USA         NY  New York City       New York   
87     closed          USA         MA         Boston        Needham   

    founded_at           ...           funding_round_code   funded_at  \
9          NaN           ...                            A  06-08-2014   
33  01-01-2012           ...                            A  26-02-2015   
69  09-05-2005           ...                            A  01-08-2006   
70  09-05-2005           ...                            B  01-09-2007   
87  01-01-2007           ...                            A  01-09-2007   

   raised_amount_usd        Country Code  primary_sector    main_sector  \
9         5000000.00  United States  USA           Media  Entertainment   
33        5000000.00  United States  USA           Media  Entertainment   
69        5000000.00  United States  USA           Media  Entertainment   
70        5000000.00  United States  USA           Media  Entertainment   
87       10000000.00  United States  USA           Media  Entertainment   

   total_amount_invested total_investment_count company_amount_invested  
9          5000212982.00                    580              5000000.00  
33         5000212982.00                    580              5000000.00  
69         5000212982.00                    580             10000000.00  
70         5000212982.00                    580             10000000.00  
87         5000212982.00                    580             10000000.00  

[5 rows x 23 columns]

In [79]:
D1A=D1A.sort_values(by=['total_investment_count' ,  'company_amount_invested' , 'permalink'] ,ascending = False )

D1A=D1A.drop_duplicates(['main_sector'], keep='first')

D1A[['permalink' ,'main_sector','total_investment_count','company_amount_invested']]


permalink  \
31816            /ORGANIZATION/VIRTUSTREAM   
40920               /ORGANIZATION/BIODESIX   
57299                /ORGANIZATION/INTACCT   
4824        /ORGANIZATION/AIRSPAN-NETWORKS   
59808  /ORGANIZATION/ENDOGASTRIC-SOLUTIONS   
25492               /ORGANIZATION/CONTROL4   
67904                 /ORGANIZATION/AVNERA   
28991       /ORGANIZATION/ZERO-MOTORCYCLES   
49685       /ORGANIZATION/ACCELERATOR-CORP   

                                   main_sector  total_investment_count  \
31816                                   Others                    2923   
40920               Cleantech / Semiconductors                    2297   
57299  Social, Finance, Analytics, Advertising                    1912   
4824                News, Search and Messaging                    1563   
59808                                   Health                     908   
25492                            Manufacturing                     799   
67904                            Entertainment                     580   
28991                      Automotive & Sports                     167   
49685                                   Blanks                      86   

       company_amount_invested  
31816              64300000.00  
40920              75300000.00  
57299              61800000.00  
4824               81769000.00  
59808              67896724.00  
25492              55700000.00  
67904              71062970.00  
28991              38124220.00  
49685              26800000.00

In [80]:
D2A=D2
D2A['company_amount_invested']= D2A.groupby(['main_sector', 'permalink'])['raised_amount_usd'].transform('sum')

In [81]:
D2A.head()

permalink          name            homepage_url  \
194          /ORGANIZATION/90MIN         90min    http://www.90min.com   
195          /ORGANIZATION/90MIN         90min    http://www.90min.com   
397        /ORGANIZATION/WAHANDA       Wahanda  http://www.wahanda.com   
3057  /ORGANIZATION/ACS-CLOTHING  ACS Clothing   http://hirewear.co.uk   
3058  /ORGANIZATION/ACS-CLOTHING  ACS Clothing   http://hirewear.co.uk   

                                          category_list     status  \
194                 Media|News|Publishing|Soccer|Sports  operating   
195                 Media|News|Publishing|Soccer|Sports  operating   
397   Beauty|Curated Web|Fitness|Health and Wellness...   acquired   
3057                                         E-Commerce  operating   
3058                                         E-Commerce  operating   

     country_code state_code   region     city  founded_at  \
194           GBR         H9   London   London  01-01-2011   
195           GBR         H9   London   London  01-01-2011   
397           GBR         H9   London   London  14-02-2008   
3057          GBR         V2  Glasgow  Glasgow  01-01-1997   
3058          GBR         V2  Glasgow  Glasgow  01-01-1997   

               ...           funding_round_code   funded_at raised_amount_usd  \
194            ...                          NaN  06-10-2015       15000000.00   
195            ...                          NaN  07-05-2013        5800000.00   
397            ...                            B  17-10-2011        8682611.00   
3057           ...                          NaN  28-09-2015       13359377.00   
3058           ...                          NaN  02-03-2014       12879637.00   

             Country Code  primary_sector    main_sector  \
194   United Kingdom  GBR           Media  Entertainment   
195   United Kingdom  GBR           Media  Entertainment   
397   United Kingdom  GBR          Beauty         Health   
3057  United Kingdom  GBR      E-Commerce         Others   
3058  United Kingdom  GBR      E-Commerce         Others   

     total_amount_invested total_investment_count company_amount_invested  
194           461511059.00                     54             20800000.00  
195           461511059.00                     54             20800000.00  
397           214537510.00                     24              8682611.00  
3057         1249124289.00                    143             26239014.00  
3058         1249124289.00                    143             26239014.00  

[5 rows x 23 columns]

In [82]:
D2A=D2A.sort_values(by=['total_investment_count' ,  'company_amount_invested' , 'permalink'] ,ascending = False )

D2A=D2A.drop_duplicates(['main_sector'], keep='first')

D2A[['permalink' ,'main_sector','total_investment_count','company_amount_invested']]

permalink  \
13837          /ORGANIZATION/ELECTRIC-CLOUD   
48255             /ORGANIZATION/EUSA-PHARMA   
18367   /ORGANIZATION/CELLTICK-TECHNOLOGIES   
7093                /ORGANIZATION/UBIQUISYS   
8420               /ORGANIZATION/MIND-CANDY   
26954                  /ORGANIZATION/NUJIRA   
61583                 /ORGANIZATION/GLYSURE   
69527  /ORGANIZATION/GREENROAD-TECHNOLOGIES   
50802      /ORGANIZATION/PARAGON-SOFTWARE-3   

                                   main_sector  total_investment_count  \
13837                                   Others                     143   
48255               Cleantech / Semiconductors                     127   
18367  Social, Finance, Analytics, Advertising                      98   
7093                News, Search and Messaging                      73   
8420                             Entertainment                      54   
26954                            Manufacturing                      42   
61583                                   Health                      24   
69527                      Automotive & Sports                      16   
50802                                   Blanks                       7   

       company_amount_invested  
13837              37000000.00  
48255              35600000.00  
18367              37500000.00  
7093               37000000.00  
8420               20860000.00  
26954              29550000.00  
61583              24412621.00  
69527              52500000.00  
50802              14000000.00

In [83]:
D3A=D3
D3A['company_amount_invested']= D3A.groupby(['main_sector', 'permalink'])['raised_amount_usd'].transform('sum')

In [84]:
D3A.head()

permalink           name              homepage_url  \
0               /ORGANIZATION/-FAME          #fame        http://livfame.com   
547  /ORGANIZATION/21DIAMONDS-INDIA     21Diamonds  http://www.21diamonds.de   
558     /ORGANIZATION/BESTYLISH-COM  beStylish.com  http://www.bestylish.com   
585         /ORGANIZATION/CARATLANE      CaratLane  http://www.caratlane.com   
586         /ORGANIZATION/CARATLANE      CaratLane  http://www.caratlane.com   

                category_list     status country_code state_code     region  \
0                       Media  operating          IND         16     Mumbai   
547                E-Commerce  operating          IND         10  New Delhi   
558  E-Commerce|Fashion|Shoes     closed          IND         10  New Delhi   
585        E-Commerce|Jewelry  operating          IND         25    Chennai   
586        E-Commerce|Jewelry  operating          IND         25    Chennai   

        city  founded_at           ...           funding_round_code  \
0     Mumbai         NaN           ...                            B   
547  Gurgaon  01-06-2012           ...                            C   
558  Gurgaon  10-05-2011           ...                            A   
585  Chennai  01-10-2008           ...                            A   
586  Chennai  01-10-2008           ...                            C   

      funded_at raised_amount_usd Country Code  primary_sector    main_sector  \
0    05-01-2015       10000000.00   India  IND           Media  Entertainment   
547  15-11-2012        6369507.00   India  IND      E-Commerce         Others   
558  01-06-2012       10000000.00   India  IND      E-Commerce         Others   
585  22-06-2011        6000000.00   India  IND      E-Commerce         Others   
586  13-05-2013       15000000.00   India  IND      E-Commerce         Others   

    total_amount_invested total_investment_count company_amount_invested  
0            280830000.00                     33             10000000.00  
547         1007409507.00                    109              6369507.00  
558         1007409507.00                    109             10000000.00  
585         1007409507.00                    109             21000000.00  
586         1007409507.00                    109             21000000.00  

[5 rows x 23 columns]

In [85]:
D3A=D3A.sort_values(by=['total_investment_count' ,  'company_amount_invested' , 'permalink'] ,ascending = False )

D3A=D3A.drop_duplicates(['main_sector'], keep='first')

D3A[['permalink' ,'main_sector','total_investment_count','company_amount_invested']]

permalink  \
622                        /ORGANIZATION/FIRSTCRY-COM   
4696   /ORGANIZATION/GUPSHUP-TECHNOLOGY-INDIA-PVT-LTD   
51276                             /ORGANIZATION/YEBHI   
14978                       /ORGANIZATION/KOMLI-MEDIA   
62162                             /ORGANIZATION/IYOGI   
35466                       /ORGANIZATION/AZURE-POWER   
59040                         /ORGANIZATION/MYDENTIST   
57095                    /ORGANIZATION/MAKEMYTRIP-COM   
49666                         /ORGANIZATION/TASKBUCKS   

                                   main_sector  total_investment_count  \
622                                     Others                     109   
4696                News, Search and Messaging                      52   
51276                            Entertainment                      33   
14978  Social, Finance, Analytics, Advertising                      32   
62162                            Manufacturing                      21   
35466               Cleantech / Semiconductors                      20   
59040                                   Health                      19   
57095                      Automotive & Sports                      13   
49666                                   Blanks                       2   

       company_amount_invested  
622                39000000.00  
4696               33000000.00  
51276              21000000.00  
14978              28000000.00  
62162              24500000.00  
35466              47200000.00  
59040              18000000.00  
57095              28000000.00  
49666              15000000.00